### PR TITLE
chore: upgrade minNodeVersion to 18+

### PR DESCRIPTION
**Pull request by kaizencc**
-----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .projenrc.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ const project = new typescript.TypeScriptProject({
   defaultReleaseBranch: 'main',
   authorName: 'Amazon Web Services',
   authorUrl: 'https://aws.amazon.com',
-  minNodeVersion: '16.17.0',
+  minNodeVersion: '18.12.0',
```
Comment (kaizencc) [inline .projenrc.ts:11]:
```suggestion
  minNodeVersion: '18.12.0',
  workflowNodeVersion: '18.x',
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -22,7 +22,7 @@ jobs:
       - name: Setup Node.js
         uses: actions/setup-node@v3
```
Comment (jfuss) [inline .github/workflows/build.yml:23]: Looks like we are duplicating work: https://github.com/cdklabs/aws-delivlib/pull/1551

You should be upgrading setup-node and checkout versions as well.

**Review by jfuss — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 2 above)
Comment (kaizencc) [reply .github/workflows/build.yml:23]: Sorry. Do you want to take over? On your PR, it looks like you manually updated the files whereas you'll need to update the `projenrc.ts` file for any changes to stick.

Comment (kaizencc) [reply .github/workflows/build.yml:23]: These versions are controlled by projen. I don't think they are necessary to fix the broken workflow atm

Comment (jfuss) [reply .github/workflows/build.yml:23]: There isn't much to take over here...

A little confused why projen is needed here but it's how it is setup.

The error (which I can't find anymore) said checkout and node actions needed upgrading to node v20. Which I don't think CDK even supports.

**Review by jfuss — APPROVED** [review]

**Pull request merged**